### PR TITLE
#159951966 Implement users profile 

### PR DESCRIPTION
**Pull request by ruganda**
#### What does this PR do?
- implement users profile
#### Description of Task to be completed?
- users should have their profiles created once they create their user accounts
-  users should retrieve their profile with the bio, image/avatar and any other fields
- Users should retrieve their profile after updating
#### How should this be manually tested?
- access `api/profiles/<username>/` to retrieve a users profile
- access `api/user/` to update or retrieve the profile while authenticated
#### What are the relevant pivotal tracker stories?
- #159951966

### CODE REVIEW

**Review by cmplx-xyttmt — CHANGES_REQUESTED** [review]
@ruganda , I think you should put the `.DS_Store` and `.vscode/settings.json` files in `.gitignore`.

**Comment by marthamareal** [comment]
@ruganda 
I appreciate the way you Implemented the feature. Good job. but,
-  it would be better to remove unnecessary files in your commit **for example,** .vscode/settings.json, migration files which are also causing duplicate issues.
- validate user profile in the serializer.
- modifying authentication with profile is out of task scope
- use the help of the serializer and jsonrenderer class to render the data you want instead of creating a dictionary in the views.py
In renderers.py attributes (charset = 'utf-8' and object_label = 'profile') refactor. charset not used and object_label used once.

**Comment by cmplx-xyttmt** [comment]
@ruganda , your commit message is the one from the previous social auth commit. Also, you still have some `.DS_Store` files.

**Comment by byarustev** [comment]
Mubarak you will need to remove migration files that are causing code climate issues. these files don't need to be tracked
You will also need to check your commit, its somehow bringing up a different message from martha's previous commit.

**Comment by ruganda** [comment]
@cmplx-xyttmt Thanks, let me work on that.

**Comment by ruganda** [comment]
@byarustev but we have migrations on version control. I can't fix that code climate issue because it is in migrations.

**Review by cmplx-xyttmt — APPROVED** [review]

**Comment by arthuroe** [comment]
@ruganda there is still a .DS_Store in your files

**Comment by arthuroe** [comment]
@ruganda conflicts 👇

**Comment by ruganda** [comment]
@arthuroe, I have resolved the conflicts

**Comment by cmplx-xyttmt** [comment]
@ruganda, you need to get rid of Stephen's changes since his commit was reverted. Your branch now shows that it has 2 commits.

**Comment by ruganda** [comment]
> @ruganda, you need to get rid of Stephen's changes since his commit was reverted. Your branch now shows that it has 2 commits.

@cmplx-xyttmt I have resolved it

**Comment by cmplx-xyttmt** [comment]
LGTM

File: authors/apps/profiles/serializers.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from rest_framework import serializers
+
+from .models import Profile
+
+
+class ProfileSerializer(serializers.ModelSerializer):
+    username = serializers.CharField(source='user.username')
+    bio = serializers.CharField(allow_blank=True, required=False)
+    # image = serializers.SerializerMethodField()
```
Comment (arthuroe) [inline authors/apps/profiles/serializers.py]: @ruganda do we need this ☝️ ?

Comment (ruganda) [reply authors/apps/profiles/serializers.py]: @arthuroe , 
- Yes we need it for retrieving a user's profile while not authenticated, or when another authenticated user is retrieving another user's profile . `api/profiles/<username>`
- `api/user` , `retrieve` and `update` can be accessed by the owner of the profile

**Comment by ruganda** [comment]
@arthuroe , I have removed the comment.

File: authors/apps/profiles/serializers.py (hunk 1 above)
Comment (ruganda) [reply authors/apps/profiles/serializers.py]: I have removed the comment. Sorry about the miss communication

**Comment by marthamareal** [comment]
LGTM

**Comment by arthuroe** [comment]
LGTM

**Pull request merged**